aioamqp/protocol.py

```diff
@@ -317,9 +321,13 @@ def channel(self, **kwargs):
         """Factory to create a new channel
 
         """
-        if not len(self.channels_ids):
-            raise exceptions.NoChannelAvailable()
-        channel_id = self.channels_ids.pop(0)
+        try:
+            channel_id = self.channels_ids_free.pop()
+        except KeyError:
+            if self.channels_ids_ceil >= amqp_constants.MAX_CHANNELS:
```

Comment:
> I don't think we will get into this branch because the server won't give us any channel with ID "MAX_CHANNEL"

I don't get it: of course we will get to this branch if we keep creating channels and not closing them... if we don't get the bounds right, the server will throw an error later